### PR TITLE
Add step by step guide for adding a data source to DLME

### PR DESCRIPTION
**Pull request by aaron-collier**
In response to #244

### CODE REVIEW

File: docs/add_data_source.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    expect(CreateResourceJob).to have_received(:perform_later) do |_id, _two, json|
+      dlme = JSON.parse(json)
+      expect(dlme['agg_provider']).to eq 'EXPECTED VALUE FROM FIXTURE'
+      ..repeat for all fields to test..
+    end
+  end
+end
+```
+
+### Mapping configuration <a name="config" />
```
Comment (cbeer) [inline docs/add_data_source.md]: Should we link to the upstream traject docs?

Comment (aaron-collier) [reply docs/add_data_source.md]: For sure. I can add that.

File: docs/add_data_source.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+### Mapping macro <a name="macro" />
+
+This step is entirely optional and dependent on the needs of the particular data
+to be translated.
+
+Traject macros provide a facility for more complex data transformation. These are
+advanced settings, [review existing macros](../lib/traject/macros) for examples of
+macro structure.
+
```
Comment (cbeer) [inline docs/add_data_source.md]: Maybe it's worth calling out the `transform_values` macro specifically, and how it is used for create the hierarchical data structures required for the `agg_is_shown_at` and `agg_is_shown_by` (any others?) keys?

File: docs/add_data_source.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+6. Assign any appropriate properties. Example:
+```
+`agg_provider`: [NAME OF DATA PROVIDER]
+`inst_id`: short id prefix [i.e. `my_source`]
+```
+
+#### Example configuration
+
+```
+stanford_mods:
```
Comment (cbeer) [inline docs/add_data_source.md]: I'd suggest grabbing the example with comments that @cmh2166 added'; in particularly, it emphasizes the optional vs required configuration.

File: docs/add_data_source.md (hunk 1 above)
Comment (aaron-collier) [reply docs/add_data_source.md]: Added link to traject project.

File: docs/add_data_source.md (hunk 2 above)
Comment (aaron-collier) [reply docs/add_data_source.md]: done.

**Pull request merged**